scripts/publish.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+
+# For local development, in production, the environment will be set though GH actions and GH secrets
+if [ -f ".envrc" ]; then
+    echo "Loading .envrc"
+    source .envrc
+else
+    echo "No .envrc"
+fi
+
+echo "============================"
+echo "PAT: $PAT"
+echo "PRODUCTION: $PRODUCTION"
+echo "============================"
+
+if [ "$PRODUCTION" = "1" ]; then
+    git config --global user.email "prismabots@gmail.com"
+    git config --global user.name "Prismo"
+    git remote add github "https://$GITHUB_ACTOR:$GH_TOKEN@github.com/$GITHUB_REPOSITORY.git" || true
+else
+    echo "Not setting up repo because PRODUCTION is not set"
+fi
+
+if [ -z "$PAT" ]; then
+    echo "\$PAT is empty. Please set the value of $PAT"
+elif [ -n "$PAT" ]; then
+    if [ "$PRODUCTION" = "1" ]; then
+        ./node_modules/.bin/vsce publish patch --pat $PAT
+    else
+        echo "Printing the command because PRODUCTION is not set"
+        echo "./node_modules/.bin/vsce publish patch --pat $PAT"
+    fi
+fi
+
+if [ "$PRODUCTION" = "1" ]; then
+    git pull github "${GITHUB_REF}" --ff-only
```

Comment:
Are there any cases where this could lead to a merge-conflict in CI?

Reply:
Also, same as above, would be useful to add a comment to say what bash is doing :-)

Reply:
> Are there any cases where this could lead to a merge-conflict in CI?

For parallel jobs maybe or manual commits to master in the duration of a CI job. But it will be fixed with the next job run and the release will happen none the less.

Reply:
> Also, same as above, would be useful to add a comment to say what bash is doing :-)

Done

Reply:
great!